libraries/chain/include/eosio/chain/unapplied_transaction_queue.hpp

```diff
@@ -37,11 +37,14 @@ struct unapplied_transaction {
 
    const transaction_id_type& id()const { return trx_meta->id(); }
    fc::time_point_sec expiration()const { return trx_meta->packed_trx()->expiration(); }
-
+#if __cplusplus >= 202002L
+   [[no_unique_address]] noncopyable _;
+#else
    unapplied_transaction(const unapplied_transaction&) = delete;
    unapplied_transaction() = delete;
    unapplied_transaction& operator=(const unapplied_transaction&) = delete;
    unapplied_transaction(unapplied_transaction&&) = default;
+#endif
```

Comment:
Instead, I think it would be better to just add the needed constructor:
```
   unapplied_transaction(transaction_metadata_ptr t, trx_enum_type type, bool return_failure_trace = false, next_func_t next = {})
   : trx_meta(std::move(t)), trx_type(type), return_failure_trace(return_failure_trace), next(std::move(next)) {}
```

Reply:
I would have to change the `insert` calls, as in:
```
         auto insert_itr = queue.insert(
               { trx, api_trx ? trx_enum_type::incoming_api : trx_enum_type::incoming_p2p, return_failure_trace, std::move( next ) } );
```
because wiith the constructor you suggest does not accept initializer lists.

Also the intention appeared to have been to disable copied (by deleting copy constructor and assignment operator).

Reply:
In my opinion modifying the insert calls or adding additional constructors or removing the disable of copy anything is better than the #ifdef.

Reply:
OK, I removed the whole `#ifdef`.